### PR TITLE
Refactors some components used in the client credential steps to be r…

### PR DESCRIPTION
**Pull request by dakota002**
…eusable, adds a new component for TimeAgo

### CODE REVIEW

**Comment by lpsinger** [comment]
We also have some dead code. Do you want to remove that in this PR too?

File: app/components/NewCredentialForm.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+import { getEnvOrDieInProduction } from '~/lib/env'
+import { ClientCredentialVendingMachine } from '~/routes/user/client_credentials.server'
+
+export async function loader({ request }: DataFunctionArgs) {
+  const machine = await ClientCredentialVendingMachine.create(request)
+  const client_credentials = await machine.getClientCredentials()
+  const groups = machine.groups
+  const recaptchaSiteKey = getEnvOrDieInProduction('RECAPTCHA_SITE_KEY')
+  return { client_credentials, recaptchaSiteKey, groups }
+}
```
Comment (lpsinger) [inline app/components/NewCredentialForm.tsx:24]: This code is duplicated.

**Review by lpsinger — CHANGES_REQUESTED** [review]
Also remove dead code from old client credentials implementation.

File: app/components/NewCredentialForm.tsx (hunk 1 above)
Comment (dakota002) [reply app/components/NewCredentialForm.tsx]: Refactored in latest commit

File: app/components/NewCredentialForm.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+import { Button, Dropdown, Label, TextInput } from '@trussworks/react-uswds'
+import { useState } from 'react'
+import ReCAPTCHA from 'react-google-recaptcha'
+import { getEnvOrDieInProduction } from '~/lib/env'
+import { ClientCredentialVendingMachine } from '~/routes/user/client_credentials.server'
+
+export async function loader({ request }: DataFunctionArgs) {
+  return await handleCredentialLoader({ request } as DataFunctionArgs)
+}
+
```
Comment (lpsinger) [inline app/components/NewCredentialForm.tsx:21]: Remove; this is dead code.

```suggestion
```

File: app/routes/quickstart/credentials.tsx [hunk 3]
```diff
@@ -6,248 +6,22 @@
...
-function getFormDataString(formData: FormData, key: string) {
-  const value = formData.get(key)
-  if (typeof value === 'string') {
-    return value
-  } else if (value === null) {
-    return undefined
-  } else {
-    throw new Response(`expected ${key} to be a string`, { status: 400 })
-  }
+  return await handleCredentialLoader({ request } as DataFunctionArgs)
```
Comment (lpsinger) [inline app/routes/quickstart/credentials.tsx:20]: The cast should not be necessary:

```suggestion
  return await handleCredentialLoader({ request })
```

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/components/NewCredentialForm.tsx (hunk 2 above)
Comment (dakota002) [reply app/components/NewCredentialForm.tsx:21]: The loader function is referenced later in the file, the returned group and the recaptcha key

File: app/components/NewCredentialForm.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+import { Form, Link, useLoaderData } from '@remix-run/react'
+import { Button, Dropdown, Label, TextInput } from '@trussworks/react-uswds'
+import { useState } from 'react'
+import ReCAPTCHA from 'react-google-recaptcha'
+import { getEnvOrDieInProduction } from '~/lib/env'
+import { ClientCredentialVendingMachine } from '~/routes/user/client_credentials.server'
+
+export async function loader({ request }: DataFunctionArgs) {
+  return await handleCredentialLoader({ request } as DataFunctionArgs)
+}
```
Comment (lpsinger) [inline app/components/NewCredentialForm.tsx:20]:
```suggestion
export async function loader(args: DataFunctionArgs) {
  return await handleCredentialLoader(args)
}
```

**Review by lpsinger — CHANGES_REQUESTED** [review]

**Review by lpsinger — APPROVED** [review]

**Pull request merged**